src/autorest-core/lib/pipeline/schema-validation.ts

```diff
@@ -23,12 +24,15 @@ export function GetPlugin_SchemaValidator(): PipelinePlugin {
     const errors = await new Promise<{ code: string, params: string[], message: string, path: string }[] | null>(res => validator.validate(obj, extendedSwaggerSchema, (err, valid) => res(valid ? null : err)));
     if (errors !== null) {
       for (const error of errors) {
+        // Replace '_' with '-' to avoid output formatter interpreting as italics
+        const errorLog = inspect(error, { depth: 3 }).replaceAll('_', '-');
+
         config.Message({
           Channel: Channel.Error,
           Details: error,
           Plugin: "schema-validator",
           Source: [{ document: fileIn.key, Position: { path: parseJsonPointer(error.path) } as any }],
-          Text: `Schema violation: ${error.message}`
+          Text: `Schema violation: ${error.message}\n${errorLog}`
```

Comment:
@timotheeguerin: Do you think we should always log this?  Or maybe only when `error.code == ONE_OF_MISSING`?

The details are certainly helpful for `ONE_OF_MISSING`, but I wonder if they might be overkill for all schema violations?

On the other hand, in the case of schema violations, probably the more information we can show the caller the better?

Reply:
In autorest v3 I have a much better log when there is schema errors but here I would say just do the cheapest unless you want to try to cherry pick those pieces